packages/cli/src/amplify.ts

```diff
@@ -3,9 +3,41 @@
 import { hideBin } from 'yargs/helpers';
 import { createMainParser } from './main_parser_factory.js';
 import { attachUnhandledExceptionListeners } from './error_handler.js';
+import { extractSubCommands } from './extract_subcommands.js';
+import {
+  AmplifyFault,
+  PackageJsonReader,
+  UsageDataEmitterFactory,
+} from '@aws-amplify/platform-core';
+import { fileURLToPath } from 'node:url';
 
-attachUnhandledExceptionListeners();
+const packageJson = new PackageJsonReader().read(
+  fileURLToPath(new URL('../package.json', import.meta.url))
+);
+const libraryVersion = packageJson.version;
 
-const parser = createMainParser();
+if (libraryVersion == undefined) {
+  throw new AmplifyFault('UnknownVersionFault', {
+    message:
+      'Library version cannot be determined. Check the library installation',
+  });
+}
 
+const usageDataEmitter = await new UsageDataEmitterFactory().getInstance(
+  libraryVersion
+);
+
+attachUnhandledExceptionListeners(usageDataEmitter);
+
+const parser = createMainParser(libraryVersion, usageDataEmitter);
 await parser.parseAsync(hideBin(process.argv));
+
+try {
+  await usageDataEmitter.emitSuccess(
+    {},
+    { command: extractSubCommands(parser) }
+  );
+  // eslint-disable-next-line amplify-backend-rules/no-empty-catch
+} catch (e) {
+  /* empty */
+}
```

Comment:
1. Can we log this in debug mode ?
2. Should this logic go to usageDataEmitter? If emitting metrics is best effort perhaps that'd be better place.

Reply:
2. is good point, so let's keep catch here.